src/main/java/org/killbill/billing/client/KillBillHttpClient.java

```diff
@@ -391,87 +391,121 @@ public Response doOptions(final String uri, final Multimap<String, String> optio
             }
         }
 
+        final Response response = request(builder, timeoutSec);
+        if (response.getStatusCode() == 404 || response.getStatusCode() == 204) {
+            return createEmptyResult(clazz);
+        }
```

Comment:
My assumption is that 404's and 204's should never have a Location header, given their nature.